pkg/virt-handler/vm.go

```diff
@@ -335,10 +335,18 @@ func (d *VirtualMachineController) execute(key string) error {
 		vmi.UID = domain.Spec.Metadata.KubeVirt.UID
 	}
 
-	// Ignore domains from an older VMI but delete watchdog file,
-	// by definition there can only be one VMI with the same name run at the same time.
 	if vmiExists && domainExists && domain.Spec.Metadata.KubeVirt.UID != vmi.UID {
-		log.Log.Object(vmi).Info("Ignoring domain from an older VMI, will be handled by its own VMI.")
+		oldVMI := v1.NewVMIReferenceFromNameWithNS(vmi.Namespace, vmi.Name)
```

Comment:
Does this imply, the watchdog files are not gone when VMI gets deleted? Where are the files stored, please?